src/debug_commads.c

```diff
@@ -665,6 +667,27 @@ DEBUG_COMMAND(GCForceBGInvoke) {
   return REDISMODULE_OK;
 }
 
+DEBUG_COMMAND(GCWaitForInvokeAndDrop) {
+  if (argc < 1) {
+    return RedisModule_WrongArity(ctx);
+  }
+  StrongRef ref = IndexSpec_LoadUnsafe(ctx, RedisModule_StringPtrLen(argv[0], NULL), 0);
+  IndexSpec *sp = StrongRef_Get(ref);
+  if (!sp) {
+    return RedisModule_ReplyWithError(ctx, "Unknown index name");
+  }
+  // wait for the timer to be called
+  uint64_t remaining;
+  if (RedisModule_GetTimerInfo(RSDummyContext, sp->gc->timerID, &remaining, NULL) == REDISMODULE_OK) {
+    usleep(1000 * remaining);
+  }
+  // Nullify the pointer so we won't try to free it in the next step.
+  // It should free itself when the timer is called and it discovers that the index was deleted.
+  sp->gc = NULL;
```

Comment:
Also, explain that we are holding the GIL here, and GC won't start running until we finish here